prow/tide/status.go

```diff
@@ -380,70 +371,29 @@ func (sc *statusController) search() []PullRequest {
 	// adding statuses to excluded repos.
 	orgExceptions, repos := sc.config().Tide.Queries.OrgExceptionsAndRepos()
 	orgs := sets.StringKeySet(orgExceptions)
-	freshOrgs, freshRepos := orgs.Difference(sc.trackedOrgs), repos.Difference(sc.trackedRepos)
-	oldOrgs, oldRepos := sc.trackedOrgs.Difference(orgs), sc.trackedRepos.Difference(repos)
-	// Stop tracking orgs and repos that aren't queried this loop.
-	sc.trackedOrgs.Delete(oldOrgs.UnsortedList()...)
-	sc.trackedRepos.Delete(oldRepos.UnsortedList()...)
-	// Determine the query for tracked PRs now before we modify 'trackedOrgs' and 'trackedRepos'.
-	var trackedQuery string
-	if sc.trackedOrgs.Len() > 0 || sc.trackedRepos.Len() > 0 {
-		trackedQuery = openPRsQuery(sc.trackedOrgs.UnsortedList(), sc.trackedRepos.UnsortedList(), orgExceptions)
-	}
-	queryStartTime := time.Now()
-
-	var lock sync.Mutex
-	var wg sync.WaitGroup
-	var allPRs []PullRequest
-	// Query fresh orgs and repos individually and since the beginning of time.
-	// These queries are larger and more likely to fail so we query for targets individually.
-	singleTargetSearch := func(query, target string, tracked sets.String) {
-		defer wg.Done()
-		searcher := newSearchExecutor(context.Background(), sc.ghc, sc.logger, query)
-		prs, err := searcher.search()
-		if err != nil {
-			sc.logger.WithError(err).Errorf("Searching for open PRs in %s.", target)
-			return
+	query := openPRsQuery(orgs.List(), repos.List(), orgExceptions)
+	start := time.Now()
+
+	prs, err := search(sc.ghc.Query, sc.logger, query, sc.latestPR, start)
+	log := sc.logger.WithFields(logrus.Fields{
+		"query":    query,
+		"duration": time.Since(start).String(),
+	})
+	log.Debugf("Found %d open PRs.", len(prs))
+	if err != nil {
+		log = log.WithError(err)
+		if len(prs) == 0 {
+			log.Error("Search failed")
+			return nil
 		}
-		func() {
-			lock.Lock()
-			defer lock.Unlock()
-			allPRs = append(allPRs, prs...)
-			tracked.Insert(target)
-		}()
-	}
-
-	wg.Add(freshOrgs.Len() + freshRepos.Len())
-	for _, org := range freshOrgs.UnsortedList() {
-		go singleTargetSearch(openPRsQuery([]string{org}, nil, orgExceptions), org, sc.trackedOrgs)
+		log.Warn("Search partially completed")
 	}
-	for _, repo := range freshRepos.UnsortedList() {
-		go singleTargetSearch(openPRsQuery(nil, []string{repo}, nil), repo, sc.trackedRepos)
+	if len(prs) > 0 {
+		sc.latestPR = prs[len(prs)-1].UpdatedAt.Time
```

Comment:
When we get partial results, are we guaranteed to get them from a specific partition of the time line?

Reply:
yes

Reply:
Thats news to me. Could you provide more details? What is the ordering?

Reply:
See L398

Reply:
https://help.github.com/en/articles/sorting-search-results